src/main/java/com/opendatasoft/elasticsearch/search/aggregations/bucket/geopointclustering/InternalGeoPointClustering.java

```diff
@@ -401,6 +405,37 @@ else if (revisit != null && ratio > 0 && neighborDistance / fixedRadius < ratio)
         }
     }
 
+    /**
+     * Computes the weighted centroid longitude, handling the antimeridian (±180°) correctly.
+     * When points are on opposite sides of the antimeridian, a simple average would place
+     * the centroid on the wrong side of the globe. This method detects such cases and
+     * adjusts the calculation accordingly.
+     */
+    private static double computeWeightedCentroidLon(double lon1, long weight1, double lon2, long weight2) {
+        long totalWeight = weight1 + weight2;
+
+        // Check if points are on opposite sides of the antimeridian
+        // If the difference is greater than 180°, they cross the antimeridian
+        double lonDiff = Math.abs(lon1 - lon2);
+        if (lonDiff > 180) {
+            // Normalize longitudes to [0, 360) for calculation
+            double normalizedLon1 = lon1 < 0 ? lon1 + 360 : lon1;
+            double normalizedLon2 = lon2 < 0 ? lon2 + 360 : lon2;
+
+            // Compute weighted average in normalized space
+            double weightedLon = (normalizedLon1 * weight1 + normalizedLon2 * weight2) / totalWeight;
```

Comment:
You should explicitly cast your long to double to avoid potential overflow. Something like:

```suggestion
            double weightedLon = (normalizedLon1 * (double) weight1 + normalizedLon2 * (double) weight2) / (double) totalWeight;
```

Reply:
Good point, thanks, but it is not necessary. In cases of double and long multiplication, java already promotes the long to a double.
There could be overflow risks in case of long and long multiplication, but it is not what we have here.
But, I can make the cast explicit for readability if you prefer?

Reply:
Yep I prefer the cast to be explicit 👍

Reply:
Done in [c93c495](https://github.com/opendatasoft/elasticsearch-aggregation-geoclustering/pull/22/commits/c93c495e5b530cbc07bc5be82859f3473d0f8947)